### PR TITLE
Feat/datasets

### PR DESCRIPTION
**Pull request by polomani**

### CODE REVIEW

File: src/resources/datasets/datasetsApi.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  /**
+   * [Delete data sets](https://doc.cognitedata.com/api/v1/#operation/deleteDataSets)
+   *
+   * ```js
+   * await client.datasets.delete([{id: 123}, {externalId: 'abc'}]);
+   * ```
+   */
+  public delete = (ids: IdEither[]) => {
+    return super.deleteEndpoint(ids);
```
Comment (polomani) [inline src/resources/datasets/datasetsApi.ts]: remove this

File: src/__tests__/resources/datasets.integration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    });
+
+    test('update', async () => {
+      const [updated] = await client.files.update([
+        { id: file.id, ...updateDataSetObject },
+      ]);
+
+      expect(updated.dataSetId).toEqual(datasets[1].id);
+    });
+  });
```
Comment (hhenriques) [inline src/__tests__/resources/datasets.integration.spec.ts:117]: We're missing a test for `datasets/aggregate`

File: src/resources/datasets/datasetsApi.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   * [Update data sets](https://docs.cognite.com/api/v1/#operation/updateDataSets)
+   *
+   * ```js
+   * const dataSets = await client.datasets.update([{id: 123, update: {description: {set: 'New description'}}}]);
+   * ```
+   */
+  public update = (changes: DataSetChange[]): Promise<DataSet[]> => {
+    return super.updateEndpoint(changes);
+  };
+}
```
Comment (hhenriques) [inline src/resources/datasets/datasetsApi.ts:79]: Missing `dataset/aggregate`

File: src/resources/classes/asset.ts [hunk 4]
```diff
@@ -53,6 +54,7 @@ export class Asset extends BaseResource<TypeAsset> implements TypeAsset {
     this.rootId = props.rootId;
     this.aggregates = props.aggregates;
     this.parentExternalId = props.parentExternalId;
+    this.dataSetId = props.dataSetId;
```
Comment (hhenriques) [inline src/resources/classes/asset.ts]: `dataSetId` needs to be added to Events, Files and Sequences

**Review by hhenriques — CHANGES_REQUESTED** [review]
Missing fields and a method on Datasets. @mrtraser here's the docs for the Datasets PR: https://pr-793.specs.preview.cogniteapp.com/v1.json.html#tag/Assets

**Review by hhenriques — APPROVED** [review]
Looks good!

**Review by mrtraser — APPROVED** [review]
LGTM, except linting ofc.

**Pull request merged**